src/libraries/Common/src/Interop/Windows/Kernel32/Interop.GetProcessName.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class Kernel32
+    {
+        [DllImport(Libraries.Kernel32, CharSet = CharSet.Unicode, EntryPoint = "QueryFullProcessImageNameW", ExactSpelling = true)]
+        private static extern bool QueryFullProcessImageName(
+            IntPtr hProcess,
+            uint dwFlags,
+            ref char lpBuffer,
+            ref uint lpdwSize);
+
+        private const int MAX_PROCESSNAME_LENGTH = 1024;
+
+        internal static string? GetProcessName(uint processId)
+        {
+            using (SafeProcessHandle h = Interop.Kernel32.OpenProcess(Interop.Advapi32.ProcessOptions.PROCESS_QUERY_LIMITED_INFORMATION, false, (int)processId))
```

Comment:
the `OpenProcess` method can fail:

https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-openprocess#return-value

we should most likely handle such case